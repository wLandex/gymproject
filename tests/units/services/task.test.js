const TimeTableService = require('../../../services/timeTable.js')
const TaskService = require('../../../services/task.js')

const taskMock = require('../mocks/entities/classes/taskMock.js')
const timeTableMock = require('../mocks/entities/classes/timeTableMock.js')

const taskService = new TaskService(timeTableMock, taskMock);
const timeTableService = new TimeTableService(timeTableMock, taskMock);


describe("Task test", () => {
  let timeTableDocument
  beforeEach(async () => {
    timeTableDocument = await timeTableService.create('Timetable 1');
  })

  //TODO 123
  test("Should create task", async () => {
    let name = "name1fsdf";
    let description = "hello dfgdfg"
    let result = await taskService.create({ttID: timeTableDocument._id, name, description})
    expect(result.timeTableID).toBe(timeTableDocument._id);
    expect(result.name).toBe(name);
    expect(result.description).toBe(description);
    expect(result._id).toBeDefined();
    expect(typeof result._id).toBe("string");
  })
  test("Should return task", () => {


  })


})