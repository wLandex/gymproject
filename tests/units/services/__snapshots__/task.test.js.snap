// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task test Should create tasks 1`] = `
{
  "_id": Any<String>,
  "description": "Some description",
  "name": "first",
  "timeTableID": Any<String>,
}
`;

exports[`Task test Should create tasks 2`] = `
{
  "_id": Any<String>,
  "description": "Some description",
  "name": "second",
  "timeTableID": Any<String>,
}
`;
